pkg/operator/ceph/object/user/controller.go

```diff
@@ -310,14 +310,20 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	}
 
 	r.objContext = objContext
-	r.objContext.Endpoint = store.Status.Info["endpoint"]
-	if store.Status != nil && store.Status.Info["secureEndpoint"] != "" {
-		r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
+	if store.Status == nil {
+		return errors.New("failed to initialize ceph object user because unknown object store status")
 	}
-	if r.objContext.Endpoint == "" {
+
+	if store.Status.Info["endpoint"] != "" {
```

Comment:
Let's check if `"endpoint"` is present in the map instead of assuming it's there.
```suggestion
        if _, ok := store.Status.Info["endpoint"] ; ok {
	    r.objContext.Endpoint = store.Status.Info["endpoint"]
	}
```